pkg/metrics/metricfactory/metric_factory.go

```diff
@@ -41,11 +43,14 @@ func EnergyMetricsPromDesc(context string) (descriptions map[string]*prometheus.
 			}
 		} else if strings.Contains(name, config.PLATFORM) && platform.IsSystemCollectionSupported() {
 			source = platform.GetSourceName()
-		} else if components.IsSystemCollectionSupported() {
+		} else if strings.Contains(allComponents, name) && components.IsSystemCollectionSupported() {
```

Comment:
can this be replaced with name != config..OTHER || name != config.UNCORE

Reply:
I guess it can be replaced with name != config.Other. Not name != config.Uncore because that is part of node components. I thought this was more clear in saying just for core, dram, pkg, uncore. I am not sure what the source label should be for other. Prior to my implementation OTHER would always be trained_power_model which I think makes sense since it is estimation by doing platform - pkg - dram. If both platform and component power was available via sensors, should Other's source still be trained_power_model?